### PR TITLE
Export quiz for moodle

### PR DESCRIPTION
**Pull request by damar-openlearning**

### CODE REVIEW

File: src/imscc/contentPackager/moodle/quiz.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { Section, Answer } from "../../types";
+import {
+  multipleChoiceQuestion,
+  multipleAnswersQuestion,
+  matchingQuestion,
+  numericalQuestion,
+  shortAnswerQuestion,
+  textOnlyQuestion,
+} from "../canvas/resource/qtiQuestion";
```
Comment (damar-openlearning) [inline src/imscc/contentPackager/moodle/quiz.ts:9]: Why import from canvas?

**Review by marulijames — APPROVED** [review]
LGTM

File: src/imscc/contentPackager/canvas/resource/quiz.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  quizTitle: string;
+  quizzes: Section[];
+}) {
+  let answerId = 1;
+  let matchingQuestionId = 1;
+  let quizContents = "";
+  for (let quiz of quizzes) {
+    switch (quiz.type) {
+      case "multiple_choice_question":
+        for (let choice of quiz.choices || []) {
```
Comment (dcollien) [inline src/imscc/contentPackager/canvas/resource/quiz.ts:27]: #codestyle

make this into a dictionary of functions, e.g.

```
const quizBuilders = {
  multiple_choice_question(quiz) {
    for (let choice of quiz.choices || []) {
      choice.id = answerId.toString();
      answerId += 1;
    }
    return multipleChoiceQuestion({ quiz });
  },
  ... etc

};

for (let quiz of quizzes) {
  quizContents += quizBuilders[quiz.type](quiz)
}
```

**Review by dcollien — APPROVED** [review]
LGTM

Code style suggestion is all

**Pull request merged**